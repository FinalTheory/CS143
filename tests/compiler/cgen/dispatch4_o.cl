-- static dispatch on another class

class Alpha inherits Beta
{
   niam() : Int {90};
};

class Beta
{
   niam() : Int {42};
   bugger() : Int {57};
};

class Main inherits IO
{
   func() : Int {(new Alpha)@Beta.niam()};
   main() : Object {{ out_int(func()); self; }};
};
