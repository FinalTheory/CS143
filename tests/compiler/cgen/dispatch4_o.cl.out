SPIM Version 6.5 of January 4, 2003
Copyright 1990-2003 by James R. Larus (larus@cs.wisc.edu).
All Rights Reserved.
See the file README for a full copyright notice.
Loaded: /usr/class/cs143/cool/lib/trap.handler
42COOL program successfully executed
Stats -- #instructions : 275
         #reads : 55  #writes 43  #branches 53  #other 124
